core/src/stores/postgres.rs

```diff
@@ -1792,43 +1797,76 @@ impl Store for PostgresStore {
             )
             .await?;
         let _ = tx
-            .query(&stmt, &[&data_source_row_id, &document_id])
+            .query(&stmt, &[&data_source_row_id, &upsert_params.document_id])
             .await?;
 
         let stmt = tx
             .prepare(
                 "INSERT INTO data_sources_documents \
                    (id, data_source, created, document_id, timestamp, tags_array, parents, \
                     source_url, hash, text_size, chunk_count, status) \
-                   VALUES (DEFAULT, $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING id",
+                   VALUES (DEFAULT, $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING id, created",
             )
             .await?;
 
-        let document_row_id = tx
+        let r = tx
             .query_one(
                 &stmt,
                 &[
                     &data_source_row_id,
                     &(document_created as i64),
-                    &document_id,
-                    &(document_timestamp as i64),
-                    &document_tags,
-                    &document_parents,
-                    &document_source_url,
-                    &document_hash,
-                    &(document_text_size as i64),
-                    &(document_chunk_count as i64),
+                    &upsert_params.document_id,
+                    &(upsert_params.timestamp as i64),
+                    &upsert_params.tags,
+                    &upsert_params.parents,
+                    &upsert_params.source_url,
+                    &upsert_params.hash,
+                    &(upsert_params.text_size as i64),
+                    &(upsert_params.chunk_count as i64),
                     &"latest",
                 ],
             )
-            .await?
-            .get(0);
+            .await?;
+
+        let document_row_id: i64 = r.get(0);
+        let created: i64 = r.get(1);
+
+        let should_upsert_node = upsert_params.title.is_some() && upsert_params.mime_type.is_some();
+
+        // TODO: defaults
```

Comment:
For now if you don't give them you don't get them. Soon will not matter as we'll start requiring them always.